packages/node/src/utils.ts

```diff
@@ -0,0 +1,26 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * Recursively read the contents of a directory.
+ *
+ * @param targetDir The directory to scan. All returned paths will be relative to this directory.
+ * @param _paths Array to hold results, passed for purposes of recursion. Not meant to be provided by the caller.
+ * @returns Array holding all relative paths
+ */
+export function deepReadDirSync(targetDir: string, _paths?: string[]): string[] {
+  const paths = _paths || [];
+  const currentDirContents = fs.readdirSync(targetDir);
+
+  currentDirContents.forEach((fileOrDirName: string) => {
+    const fileOrDirAbsPath = path.join(targetDir, fileOrDirName);
+
+    if (fs.statSync(fileOrDirAbsPath).isDirectory()) {
+      deepReadDirSync(fileOrDirAbsPath, paths);
+      return;
+    }
+    paths.push(fileOrDirAbsPath.replace(targetDir, ''));
+  });
+
+  return paths;
+}
```

Comment:
```suggestion
/**
 * Recursively read the contents of a directory.
 *
 * @param targetDir The directory to scan. All returned paths will be relative to this directory.
 * @returns Array holding all relative paths
 */
export function deepReadDirSync(targetDir: string): string[] {
  return fs
    .readdirSync(targetDir)
    .map((fileOrDirName: string) => {
      const fileOrDirAbsPath = path.join(targetDir, fileOrDirName);

      if (fs.statSync(fileOrDirAbsPath).isDirectory()) {
        return deepReadDirSync(fileOrDirAbsPath);
      }

      return fileOrDirAbsPath.replace(targetDir, "");
    })
    .flat();
}
```

Reply:
Or if we need to support Node 10:

```suggestion
/**
 * Recursively read the contents of a directory.
 *
 * @param targetDir The directory to scan. All returned paths will be relative to this directory.
 * @returns Array holding all relative paths
 */
export function deepReadDirSync(targetDir: string): string[] {
  return fs
    .readdirSync(targetDir)
    .reduce((acc: string[], fileOrDirName: string) => {
      const fileOrDirAbsPath = path.join(targetDir, fileOrDirName);

      if (fs.statSync(fileOrDirAbsPath).isDirectory()) {
        return [...acc, ...deepReadDirSync(fileOrDirAbsPath)];
      }

      return [...acc, fileOrDirAbsPath.replace(targetDir, "")];
    }, []);
}
```

Reply:
Ended up reworking it based on your second solution.